transport-native-kqueue/src/main/c/netty_kqueue_native.c

```diff
@@ -259,14 +260,62 @@ static jshort netty_kqueue_native_noteDisconnected(JNIEnv* env, jclass clazz) {
    return NOTE_DISCONNECTED;
 }
 
-static jint netty_kqueue_bsdsocket_connectResumeOnReadWrite(JNIEnv *env) {
+static jint netty_kqueue_bsdsocket_connectResumeOnReadWrite(JNIEnv *env, jclass clazz) {
     return CONNECT_RESUME_ON_READ_WRITE;
 }
 
-static jint netty_kqueue_bsdsocket_connectDataIdempotent(JNIEnv *env) {
+static jint netty_kqueue_bsdsocket_connectDataIdempotent(JNIEnv *env, jclass clazz) {
     return CONNECT_DATA_IDEMPOTENT;
 }
 
+static jint netty_kqueue_bsdsocket_fastOpenClient(JNIEnv *env, jclass clazz) {
+#if __APPLE__
+    unsigned int value = 0;
+    size_t len = sizeof(value);
+    if (sysctlbyname("net.inet.tcp.fastopen", &value, &len, NULL, 0) != 0) {
+        netty_unix_errors_throwRuntimeExceptionErrorNo(env, "sysctlbyname failed", errno);
+    }
+    if ((value & 2) == 2) {
+        return 1;
+    } else {
+        return 0;
+    }
+#else
+    unsigned int value = 0;
+    size_t len = sizeof(value);
+    if (sysctlbyname("net.inet.tcp.fastopen.client_enable", &value, &len, NULL, 0) != 0) {
+        netty_unix_errors_throwRuntimeExceptionErrorNo(env, "sysctlbyname failed", errno);
+    }
+    if (value) {
+        return 1;
+    } else {
+        return 0;
+    }
+#endif // __APPLE__
+}
+
+static jint netty_kqueue_bsdsocket_fastOpenServer(JNIEnv *env, jclass clazz) {
+#if __APPLE__
```

Comment:
nit: usually we use `#ifdef` when possible
```suggestion
#if __APPLE__
```